### PR TITLE
[MODORDERS-1181] Extend mapping params with ref data necessary for Orders mappings

### PR DESCRIPTION
**Pull request by RomanChernetskyi**
### Purpose 
Extend mapping params with ref data necessary for Orders mappings
### Approach
Add next fields to the mapping params:
- Locations
- MaterialTypes
- ContributorNameTypes
- Funds
- ExpenseClasses
- IdentifierTypes
- AcquisitionsUnits
- AcquisitionMethods
- TenantConfigurationAddresses
#### Jira
https://folio-org.atlassian.net/browse/MODORDERS-1181

### CODE REVIEW

**Review by KaterynaSenchenko — APPROVED** [review]

File: src/main/java/org/folio/service/caches/MappingParametersCache.java [hunk 1]
```diff
@@ -70,64 +108,87 @@ public MappingParametersCache(Vertx vertx, RestClient restClient) {
...
+    Future<List<Location>> locationsFuture = getLocations(params);
+    Future<List<Mtype>> materialTypesFuture = getMaterialTypes(params);
+    Future<List<ContributorNameType>> contributorNameTypesFuture = getContributorNameTypes(params);
+    Future<List<Organization>> organizationsFuture = getOrganizations(params);
+    Future<List<Fund>> fundsFuture = getFunds(params);
+    Future<List<ExpenseClass>> expenseClassesFuture = getExpenseClasses(params);
+    Future<List<IdentifierType>> identifierTypesFuture = getIdentifierTypes(params);
+    Future<List<AcquisitionsUnit>> acquisitionsUnitsFuture = getAcquisitionsUnits(params);
+    Future<List<AcquisitionMethod>> acquisitionsMethodsFuture = getAcquisitionMethods(params);
+    Future<List<String>> tenantConfigurationAddressesFuture = getTenantConfigurationAddresses(params);
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/caches/MappingParametersCache.java:139]: seems knowing exacty type of variable is not much important here, as a option we can use var it make it this block of code more easier to read
```

   var locationsFuture = getLocations(params);
    var materialTypesFuture = getMaterialTypes(params);
    var contributorNameTypesFuture = getContributorNameTypes(params);
    var organizationsFuture = getOrganizations(params);
    var fundsFuture = getFunds(params);
    var expenseClassesFuture = getExpenseClasses(params);
    var identifierTypesFuture = getIdentifierTypes(params);
    var acquisitionsUnitsFuture = getAcquisitionsUnits(params);
    var acquisitionsMethodsFuture = getAcquisitionMethods(params);
    var tenantConfigurationAddressesFuture = getTenantConfigurationAddresses(params);
```

File: src/main/java/org/folio/service/caches/MappingParametersCache.java [hunk 2]
```diff
@@ -70,64 +108,87 @@ public MappingParametersCache(Vertx vertx, RestClient restClient) {
...
 
     return RestUtil.doRequest(params, getOrganizationsSortedLimitPath(settingsLimit), HttpMethod.GET, null)
-      .toCompletionStage()
-      .toCompletableFuture()
-      .thenCompose(httpResponse -> {
+      .compose(httpResponse -> {
         if (httpResponse.getResponse().statusCode() == HttpStatus.SC_OK) {
           OrganizationCollection orgCollection = Json.decodeValue(httpResponse.getJson().encode(), OrganizationCollection.class);
-          LOGGER.info("loadMappingParameters:: The first chunk of organizations was loaded for cache, tenantId '{}', organizations '{}', totalRecords '{}'",
+          LOGGER.debug("loadMappingParameters:: The first chunk of organizations was loaded for cache, tenantId '{}', organizations '{}', totalRecords '{}'",
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/caches/MappingParametersCache.java:166]: also should getOrganizations 
LOGGER.debug("getOrganizations::......

File: src/main/java/org/folio/service/caches/MappingParametersCache.java [hunk 3]
```diff
@@ -145,6 +206,115 @@ private List<Future<OrganizationCollection>> getOrganizationCollectionFutures(Ma
     return organizationCollectionFutures;
   }
 
+  private Future<List<Mtype>> getMaterialTypes(OkapiConnectionParams params) {
+    String materialTypesUrl = "/material-types?limit=" + settingsLimit;
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/caches/MappingParametersCache.java:210]: using StringBuilder is more effecient for add operations

File: src/main/java/org/folio/service/caches/MappingParametersCache.java [hunk 4]
```diff
@@ -145,6 +206,115 @@ private List<Future<OrganizationCollection>> getOrganizationCollectionFutures(Ma
...
+        .stream().map(method -> mapTo(method, AcquisitionMethod.class)).toList());
+  }
+
+  private static <T> T mapTo(Object unit, Class<T> objectClass) {
+    return Json.decodeValue(Json.encode(unit), objectClass);
+  }
+
+  private Future<List<String>> getTenantConfigurationAddresses(OkapiConnectionParams params) {
+    return RestUtil.doRequest(params, TENANT_CONFIGURATION_ADDRESSES_URL, HttpMethod.GET, null).compose(httpResponse -> {
+      if ((httpResponse.getResponse().statusCode() == HttpStatus.SC_OK)) {
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/caches/MappingParametersCache.java:263]: suggested to change to `if (httpResponse.getResponse().statusCode() == HttpStatus.SC_OK) {`

File: src/main/java/org/folio/service/caches/MappingParametersCache.java [hunk 5]
```diff
@@ -145,6 +206,115 @@ private List<Future<OrganizationCollection>> getOrganizationCollectionFutures(Ma
...
+          }
+          return Future.succeededFuture(Collections.emptyList());
+        } else {
+          String message = format(ERROR_LOADING_CACHE_MESSAGE, parameterName, params.getTenantId(),
+            httpResponse.getResponse().statusCode(), httpResponse.getBody());
+          LOGGER.warn("loadData:: {}", message);
+          return Future.failedFuture(new CacheLoadingException(message));
+        }
+      } catch (Exception e) {
+        LOGGER.warn("loadData:: Failed to load {}", dataCollectionField, e);
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/caches/MappingParametersCache.java:306]: log.error is more suitable for exceptions

File: src/main/java/org/folio/service/caches/MappingParametersCache.java [hunk 6]
```diff
@@ -39,8 +63,17 @@ public class MappingParametersCache {
...
   private static final String ORGANIZATIONS = "/organizations/organizations";
-
   private static final String SORT_BY_ID_QUERY = "(cql.allRecords=1) sortBy id";
+  private static final String LOCATIONS_RESPONSE_PARAM = "locations";
+  private static final String MATERIALS_TYPES_RESPONSE_PARAM = "mtypes";
+  private static final String CONTRIBUTOR_NAME_TYPES_RESPONSE_PARAM = "contributorNameTypes";
+  private static final String FUNDS_RESPONSE_PARAM = "funds";
+  private static final String EXPENSE_CLASSES_RESPONSE_PARAM = "expenseClasses";
+  private static final String IDENTIFIER_TYPES_RESPONSE_PARAM = "identifierTypes";
+  private static final String TENANT_CONFIGURATION_ADDRESSES_URL = "/configurations/entries?query=" + URLEncoder.encode("(module==TENANT and configName==tenant.addresses)", StandardCharsets.UTF_8);
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/caches/MappingParametersCache.java:73]: too long line, it makes hard to read

**Review by dmytrokrutii — APPROVED** [review]

File: src/main/java/org/folio/service/caches/MappingParametersCache.java (hunk 1 above)
Comment (RomanChernetskyi) [reply src/main/java/org/folio/service/caches/MappingParametersCache.java]: Done, thank you

File: src/main/java/org/folio/service/caches/MappingParametersCache.java (hunk 4 above)
Comment (RomanChernetskyi) [reply src/main/java/org/folio/service/caches/MappingParametersCache.java]: Done, thank you

File: src/main/java/org/folio/service/caches/MappingParametersCache.java (hunk 6 above)
Comment (RomanChernetskyi) [reply src/main/java/org/folio/service/caches/MappingParametersCache.java]: refactored, thank you

File: src/main/java/org/folio/service/caches/MappingParametersCache.java (hunk 3 above)
Comment (RomanChernetskyi) [reply src/main/java/org/folio/service/caches/MappingParametersCache.java:212]: I think using StringBuilder will not make code more effective, but will also make it harder to read

File: src/main/java/org/folio/service/caches/MappingParametersCache.java (hunk 5 above)
Comment (RomanChernetskyi) [reply src/main/java/org/folio/service/caches/MappingParametersCache.java:313]: according to the folio documentation error level logging should be used for unhandled errors
https://folio-org.atlassian.net/wiki/spaces/DD/pages/1779260/Logging

Comment (azizbekxm) [reply src/main/java/org/folio/service/caches/MappingParametersCache.java:313]: ok. thanks to share

File: src/main/java/org/folio/service/caches/MappingParametersCache.java (hunk 3 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/caches/MappingParametersCache.java:212]: since using small concatenate, make sense to keep as it was

**Review by Aliaksandr-Fedasiuk — APPROVED** [review]

**Review by RuslanLavrov — APPROVED** [review]

**Pull request merged**